src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -160,6 +160,13 @@ public Instance(
         Object maxReturnedMetrics = this.instanceMap.get("max_returned_metrics");
         if (maxReturnedMetrics == null) {
             this.maxReturnedMetrics = MAX_RETURNED_METRICS;
+        } else if (maxReturnedMetrics instanceof String) {
+            try {
+                this.maxReturnedMetrics = Integer.parseInt((String) maxReturnedMetrics);
+            } catch (NumberFormatException e) {
+                log.warn("Cannot convert max_returned_metrics to integer in your instance configuration. Defaulting to '{}'.", MAX_RETURNED_METRICS);
```

Comment:
This line might break the linter (waiting for it to run) so you may want to do something like this (I probably have the spaces all wrong):
```suggestion
                log.warn(
                    "Cannot convert max_returned_metrics to integer in your instance configuration. Defaulting to '{}'.",
                     MAX_RETURNED_METRICS);
```